docs/DevEnvironment.md

```diff
@@ -0,0 +1,158 @@
+going to need one to contribute your improvements back to the Rouge community.
+Creating [an account][gh-make-acc] is free of charge. And just think how awesome
+it'll be when you take your collaboration to the "next level".
+
+[gh-make-acc]: https://github.com/join
+
+## Getting Rouge
+
+### Forking the Repository
+
+To develop Rouge, we're going to create a "fork" of Rouge where we can make our
+changes. These will be happily isolated from everyone else and then—through the
+magic of Git—mergeable back into the main project later.
+
+First, visit [the front page][rouge-fp] of Rouge's repo on GitHub. Unless you're
+crazy and are trying to develop code on your phone (like perhaps I am doing
+right now), you'll see a button near the top right of your browser window that
+will say "Fork". Click this and GitHub will ask where you want to create your
+fork. Select your account and—boom!—you've just forked Rouge.
+
+[rouge-fp]: https://github.com/rouge-ruby/rouge
+
+### Cloning Your Fork
+
+The next thing to do is to get your fork onto your computer. Git makes this
+easy. In the directory you want to hold your repository, type:
+
+```shell
+git clone git@github.com:<your_github_account_name>/rouge.git
+```
+
+Git will reach out to GitHub, grab the code and put it in a directory called
+`rouge/`.
+
+## Installing Development Dependencies
+
+Ruby provides support for using external "packages" of Ruby code. These packages
+are called _gems_. While Rouge itself does not depend on any gems to perform
+syntax highlighting, it does have a number of dependencies that you need to
+install to _develop_ Rouge. These are the development dependencies.
+
+### Installing Bundler
+
+The easiest way to install Rouge's development dependencies is using
+[Bundler][]. Bundler is itself a gem and we'll install it differently to how we
+install the other dependencies.
+
+[Bundler]: https://bundler.io/
+
+If you already develop with Ruby, you no doubt have Bundler installed. You can
+check if you do by typing `bundle -v` at the command line. If you don't see the
+version number then you need to install Bundler. To do this, type:
+
+```shell
+gem install bundler
+```
+
+Ruby's `gem` tool will grab the latest Bundler package and install it. Once
+that's complete, you're ready to rock.
+
+### Installing the Dependencies
+
+Rouge comes with a list of gems it depends upon called a _Gemfile_. Make sure
+you're at the top level of your clone of your repository and type:
+
+```shell
+bundle install --path vendor
+```
+
+This command tells Bundler to install the gems Rouge in the Gemfile into a
+directory called `vendor/`. This has one drawback (explained below) but means
+the gems we use for Rouge are isolated from the other gems we may have installed
+on our system. This will be tremendously helpful in avoiding conflicts that
+arise because of the use of incompatible versions of a gem.
+
+The one drawback is that we will need to tell Ruby every time we run our Rouge
```

Comment:
Users on Unix-like OS may configure their BASH terminals to *alias* `bundle exec` to something shorter like `be`..
So they can run `be rake`

Reply:
This is what I do but I don't think is necessary for the purposes of this guide. If you're comfortable enough setting aliases, you already know you can do this. For more novice users, I fear this is confusing (particularly when you start needing to explain where they need to define these aliases so they don't disappear on the next login).